src/YamlFrontMatter.php

```diff
@@ -23,6 +23,45 @@ public static function parse(string $content): Document
         return new Document($matter, $body);
     }
 
+    /**
+     * A parser that can handle Markdown that contains Markdown.
+     *
+     * Fixes https://github.com/spatie/yaml-front-matter/discussions/30.
+     * Original code by https://github.com/eklausme
+     *
+     * @param string $content
+     * @return Document
+     */
+    public static function markdownCompatibleParse(string $content): Document
```

Comment:
This code does not meet our quality expectations.

- Do not use abbreviations
- Always use strict checking
- `if` blocks should always have a body
- if you have to use a comment, there's a high change it could be replace with a function with a clear name

Could you refactor this code to be as readable as possible?

Reply:
Yeah I will give it a shot! Can't promise I'll be able to get it done today though.